### ISSUE TITLE
[no-empty-function] disable for decorators

### ISSUE THREAD

**Issue by vegerot** [typescript-eslint/typescript-eslint#2278]
**Repro**
```ts
@Emit('click')
private onClick() {}
```

Proposed option: 
```JSON
{
  "rules": {
    "@typescript-eslint/no-empty-function": ["error", {decorators: 'off'}]
  }
}
```

```TS
@Emit('click')
private onClick() {}
```

**Expected Result**
No error
**Actual Result**
Error
**Additional Info**

**Versions**

| package                            | version |
| ---------------------------------- | ------- |
| `@typescript-eslint/eslint-plugin` | `3.5.0` |
| `@typescript-eslint/parser`        | `3.5.0` |

**Comment by bradzacher** [comment]
happy to accept a PR if you want to add a new option `"decoratedFunctions"` to the `allow` option.

**Comment by vegerot** [comment]
I'd love to.  Do you have any advice for getting started?

**Comment by bradzacher** [comment]
The rule source is here.

https://github.com/typescript-eslint/typescript-eslint/blob/master/packages/eslint-plugin/src/rules/no-empty-function.ts

It is built on top of the base eslint rule which you can find in their repo.

You can use https://astexplorer.net to inspect the ast and figure out how to build the rule.
